specs/protocol/holocene/configurability.md

```diff
@@ -0,0 +1,89 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Configurability](#configurability)
+  - [Overview](#overview)
+  - [Constants](#constants)
+    - [`ConfigType`](#configtype)
+  - [`SystemConfig`](#systemconfig)
+    - [Interface](#interface)
+  - [`OptimismPortal`](#optimismportal)
+    - [Interface](#interface-1)
+      - [`setConfig`](#setconfig)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Configurability
+
+## Overview
+
+The `SystemConfig` and `OptimismPortal` are updated with a new flow for chain
+configurability. This new flow reduces the need for the `ConfigUpdate` events
+emitted by the `SystemConfig`, which adds great cost to the fault proof program's
+execution as it results in the program to iterate through all possible logs to
+ensure that it has observed all of the events. It also enables the L1 Attributes
+transaction to not post duplicate data between transactions which reduces database
+size growth over time.
+
+## Constants
+
+### `ConfigType`
+
+The `ConfigType` enum represents configuration that can be modified.
+
+| Name | Value | Description |
+| ---- | ----- | --- |
+| `SET_GAS_LIMIT` | `uint8(0)` | Sets the L2 gas limit |
+| `SET_FEE_SCALARS` | `uint8(1)` | Sets the L2 fee configuration |
+| `SET_EIP_1559_PARAMS` | `uint8(2)` | Sets the EIP-1559 elasticity and denominator |
+| `SET_BATCHER_HASH` | `uint8(3)` | Sets the batcher role |
+| `SET_UNSAFE_BLOCK_SIGNER` | `uint8(4)` | Sets the p2p sequencer key |
+| `SET_GAS_PAYING_TOKEN` | `uint8(5)` | Modifies the gas paying token for the chain |
+| `SET_BATCH_INBOX` | `uint8(6)` | Sets the canonical batch inbox address |
+
+TODO: make sure that `SET_BATCH_INBOX` doesn't break the fault proof
+
+## `SystemConfig`
+
+### Interface
+
+Each listed function calls `OptimismPortal.setConfig(ConfigType)` with its corresponding
+`ConfigType`.
+
+- `setGasLimit(uint64)`
+- `setFeeScalars(uint32,uint32)`
+- `setEIP1559Params(uint64,uint64)`
+- `setBatcherHash(bytes32)`
+- `setUnsafeBlockSigner(address)`
+- `setGasPayingToken(address)`
+- `setBatchInbox(address)`
+
```

Comment:
@roberto-bayardo @blockchaindevsh  What was the API that we wanted to standardize on?

Reply:
It was `setFeeScalars`, more specifically how we want to implement `StaticConfig.encodeSetFeeScalars`. I think for the Holocene upgrade we still want to maintain the [scalar](https://github.com/ethereum-optimism/optimism/blob/d7e5418745ef5da2df999a98da11cde228a4fbaa/packages/contracts-bedrock/src/L1/SystemConfigHolocene.sol#L47) for backwards compatibility, that's why I implemented `StaticConfig.encodeSetFeeScalars` by encoding the calculated scalar: [source](https://github.com/blockchaindevsh/optimism/blob/e876fe4029ddac58df548661fdc892267b975004/packages/contracts-bedrock/src/L1/SystemConfigHolocene.sol#L50). Do we want to implement it as ```solidity
StaticConfig.encodeSetFeeScalars(uint32 _basefeeScalar, uint32 _blobBasefeeScalar) internal pure returns (bytes memory)  {
    return abi.encode(_basefeeScalar, _blobBasefeeScalar);
}
``` instead ?

Reply:
I would prefer to make no diff to this and just leave the most significant byte as a version (0 or 1) as to reduce need for diff in the derivation pipeline.